From cb0310d39023917667db8e9b4b665b11e20e9049 Mon Sep 17 00:00:00 2001
From: Roman Shaposhnik <rvs@cloudera.com>
Date: Tue, 17 Sep 2013 14:07:17 -0700
Subject: [PATCH 50/56] CDH-14276. let flume autodetect search dependencies just as it autodetects hadoop/hbase

---
 bin/flume-ng |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/bin/flume-ng b/bin/flume-ng
index be3f80b..6ffff09 100755
--- a/bin/flume-ng
+++ b/bin/flume-ng
@@ -163,6 +163,15 @@ add_HBASE_paths() {
 
   fi
 }
+add_search_paths() {
+  local SEARCH_LIBS=${SEARCH_HOME:-${FLUME_HOME}/../search}/lib
+
+  if [ -d "$SEARCH_LIBS" ]; then
+    for PIECE in $SEARCH_LIBS/*.jar ; do
+      FLUME_CLASSPATH="$FLUME_CLASSPATH:$PIECE"
+    done
+  fi
+}
 
 set_LD_LIBRARY_PATH(){
 #Append the FLUME_JAVA_LIBRARY_PATH to whatever the user may have specified in
@@ -424,6 +433,7 @@ fi
 # look for hadoop libs
 add_hadoop_paths
 add_HBASE_paths
+add_search_paths
 
 # prepend conf dir to classpath
 if [ -n "$opt_conf" ]; then
-- 
1.7.0.4

